### PR TITLE
Create Jira tickets on Databricks for pull request reviews (#20)

### PR DESCRIPTION
**Pull request by pogil**
Defines a github action to submit a Jira ticket at Databricks' ML Platform team board to review the open source pull request. The secrets and github actions need to be configured separately for the github repository.

## What changes are proposed in this pull request?

Add a github action workflow for PR creation that calls Jira REST API to create a ticket we can use in Databricks to monitor reviews of open source contributions.

## How is this patch tested?

I set this up in pogil/mlflow and verified that it creates appropriate tickets in Jira.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by sueann — APPROVED** [review]
lgtm - thank you!

File: .github/workflows/jira.pr.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+on: 
+  pull_request:
+    branches: 
+    - master
+
+name: Jira Ticket Creation for PRs
```
Comment (harupy) [inline .github/workflows/jira.pr.yml]: This might be a nit. `Jira` -> `JIRA`.

Comment (pogil) [reply .github/workflows/jira.pr.yml]: It turns out Atlassian uses "Jira" instead of "JIRA", so it is my PR description that's flawed.

Comment (pogil) [reply .github/workflows/jira.pr.yml]: Except that I use both JIRA and Jira in the workflow 🤦‍♂

**Pull request merged**